### ISSUE TITLE
Mail address email constructor argument type incorrect

### ISSUE THREAD

**Issue by DuckThom** [magento/magento2#25434]
### Preconditions (*)
Magento 2.3.3

### Steps to reproduce (*)
```
>>> $address = new \Magento\Framework\Mail\Address(null, null);
=> Magento\Framework\Mail\Address {#17559}
>>> $address->getName();
=> null
>>> $address->getEmail();
TypeError: Return value of Magento/Framework/Mail/Address::getEmail() must be of the type string, null returned
```

### Expected result (*)
Either one of:
- Instantiating the `\Magento\Framework\Mail\Address` class with a `null` value in the email parameter should not be possible, or;
- the `getEmail` method's return type should be nullable, the same as the constructor argument ( `?string` )

### Actual result (*)
Instantiating the Address class with a `null` email address is possible which will always cause the getEmail method to throw an exception as it's only allowed to return `string` values

```
TypeError: Return value of Magento/Framework/Mail/Address::getEmail() must be of the type string, null returned
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22783` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by VladimirZaets** [comment]
Hi @DuckThom. Thank you for your report. 
The issue has been fixed in magento/magento2#25485 by @karyna-tsymbal-atwix in 2.3-develop branch
Related commit(s):
 - [286840303d1558b65cf50017b00866f89ba297ec](https://github.com/magento/magento2/commit/286840303d1558b65cf50017b00866f89ba297ec)
 - [bd44b0b03de7b05dcba98b7e0dc087ecf1535386](https://github.com/magento/magento2/commit/bd44b0b03de7b05dcba98b7e0dc087ecf1535386)
 - [d40fbd907889c5827f0fb85abc0f9f1cfddd2ab2](https://github.com/magento/magento2/commit/d40fbd907889c5827f0fb85abc0f9f1cfddd2ab2)

The fix will be available with the upcoming 2.3.5 release.
